core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -135,7 +135,7 @@
     private DataSize filterAndProjectMinOutputPageSize = DataSize.of(500, KILOBYTE);
     private int filterAndProjectMinOutputPageRowCount = 256;
     private int maxGroupingSets = 2048;
-    private JoinPushdownMode joinPushdownMode = JoinPushdownMode.DISABLED;
+    private JoinPushdownMode joinPushdownMode = JoinPushdownMode.AUTOMATIC;
```

Comment:
see conversation about code level documentation in the other pr

Reply:
Added comment as a separate commit before introducing `AUTOMATIC` mode.